pandas/core/apply.py

```diff
@@ -158,19 +159,54 @@ def agg(self) -> DataFrame | Series | None:
             return self.apply_str()
 
         if is_dict_like(arg):
-            return self.agg_dict_like()
+            return self.dict_like("agg")
         elif is_list_like(arg):
             # we require a list, but not a 'str'
-            return self.agg_list_like()
+            return self.list_like("agg")
 
         if callable(arg):
             f = com.get_cython_func(arg)
             if f and not args and not kwargs:
                 return getattr(obj, f)()
+            elif not isinstance(obj, SelectionMixin):
+                # i.e. obj is Series or DataFrame
+                return self.agg_udf()
 
         # caller can react
         return None
 
+    def agg_udf(self):
```

Comment:
What is "udf" here a shorthand for? Can you rename to something more explicit and/or add a doc string & type hints to help better understand this?

Reply:
User Defined Function; I'd prefer sticking with the name but completely agree it should be expanding upon in a docstring.

Reply:
👍